tests/cli/test_status_diff.py

```diff
@@ -0,0 +1,1188 @@
+            self.assertEqual(out, expected)
+
+    def test_macaddress_diff(self):
+        input_data = {'netplan-global-state': {'online': True, 'nameservers': {'addresses': ['127.0.0.53'], 'search': ['lxd'], 'mode': 'stub'}}, 'lo': {'index': 1, 'adminstate': 'UP', 'operstate': 'UNKNOWN', 'type': 'ethernet', 'macaddress': '00:00:00:00:00:00', 'addresses': [{'127.0.0.1': {'prefix': 8}}, {'::1': {'prefix': 128}}], 'routes': [{'to': '127.0.0.0/8', 'family': 2, 'from': '127.0.0.1', 'type': 'local', 'scope': 'host', 'protocol': 'kernel', 'table': 'local'}, {'to': '127.0.0.1', 'family': 2, 'from': '127.0.0.1', 'type': 'local', 'scope': 'host', 'protocol': 'kernel', 'table': 'local'}, {'to': '127.255.255.255', 'family': 2, 'from': '127.0.0.1', 'type': 'broadcast', 'scope': 'link', 'protocol': 'kernel', 'table': 'local'}, {'to': '::1', 'family': 10, 'metric': 256, 'type': 'unicast', 'scope': 'global', 'protocol': 'kernel', 'table': 'main'}, {'to': '::1', 'family': 10, 'type': 'local', 'scope': 'global', 'protocol': 'kernel', 'table': 'local'}]}, 'enp5s0': {'index': 2, 'adminstate': 'UP', 'operstate': 'UP', 'type': 'ethernet', 'backend': 'networkd', 'id': 'enp5s0', 'macaddress': '00:16:3e:71:d0:1f', 'vendor': 'Red Hat, Inc.', 'addresses': [{'10.86.126.148': {'prefix': 24, 'flags': ['dhcp']}}], 'dns_addresses': ['10.86.126.1'], 'dns_search': ['lxd'], 'routes': [{'to': 'default', 'family': 2, 'via': '10.86.126.1', 'from': '10.86.126.148', 'metric': 100, 'type': 'unicast', 'scope': 'global', 'protocol': 'dhcp', 'table': 'main'}, {'to': '10.86.126.0/24', 'family': 2, 'from': '10.86.126.148', 'metric': 100, 'type': 'unicast', 'scope': 'link', 'protocol': 'kernel', 'table': 'main'}, {'to': '10.86.126.1', 'family': 2, 'from': '10.86.126.148', 'metric': 100, 'type': 'unicast', 'scope': 'link', 'protocol': 'dhcp', 'table': 'main'}, {'to': '10.86.126.148', 'family': 2, 'from': '10.86.126.148', 'type': 'local', 'scope': 'host', 'protocol': 'kernel', 'table': 'local'}, {'to': '10.86.126.255', 'family': 2, 'from': '10.86.126.148', 'type': 'broadcast', 'scope': 'link', 'protocol': 'kernel', 'table': 'local'}, {'to': 'ff00::/8', 'family': 10, 'metric': 256, 'type': 'multicast', 'scope': 'global', 'protocol': 'kernel', 'table': 'local'}]}}  # nopep8
+        state_diff = {'interfaces': {'enp5s0': {'index': 2, 'name': 'enp5s0', 'id': 'enp5s0', 'system_state': {'missing_macaddress': 'aa:bb:cc:dd:ee:ff'}, 'netplan_state': {'missing_macaddress': '00:16:3e:71:d0:1f'}}}, 'missing_interfaces_system': {}, 'missing_interfaces_netplan': {'lo': {'type': 'ethernet', 'index': 1}}}  # nopep8
+
+        expected = '''+ ●  1: lo ethernet UNKNOWN/UP (unmanaged)
+          MAC Address: 00:00:00:00:00:00
+            Addresses: 127.0.0.1/8
+                       ::1/128
+               Routes: ::1 metric 256
+
+  ●  2: enp5s0 ethernet UP (networkd: enp5s0)
++         MAC Address: 00:16:3e:71:d0:1f (Red Hat, Inc.)
+-                      aa:bb:cc:dd:ee:ff (Red Hat, Inc.)
```

Comment:
parise: Thanks for adding the +/- signs, so we can test this code even without the color theming.